python/lsst/ts/cbp/component.py

```diff
@@ -192,6 +229,8 @@ async def move_azimuth(self, position: float):
         if position < -45 or position > 45:
             raise ValueError("New azimuth value exceeds Azimuth limit.")
         else:
+            self.azimuth_target = position
+            self.in_motion = True
```

Comment:
Also consider caching some of this data in the event you use to report it.

Advantages include:
* Changes are easily tracked and reported.
* There is no danger of one cache (in the model) going out of date with another (in the CSC).
* No need to call back to the CSC every time the internal state changes.
* Such callbacks are likely expensive because the CSC has to assume anything might have changed, whereas the model knows exactly what has changed.
Disadvantages:
* The modularity suffers. The SAL events leak in to the model. I often handle this by making the model contain a copy of the CSC, so it can get to the events. It's circular, but it works.

For example if your CSC has an axesInPostion event then you could
```
self.csc.evt_axesInPosition.set_put(azimuth=False, force_output=True)
```
To ask for the most recently reported value (which is not necessarily the same as the actual current value, but should be exactly the same as any other cached value, such as from self.encoder_motion):
```
azimuth_in_position = self.csc.evt.axesInPosition.data.azimuth
```
And of course you have to poll the actuators or something to keep the data up to date. The polling loop would set:
```
azimuth_in_position = ... as above
elevation_in_position = ...
self.csc.evt_axesInPosition.set_put(
    azimuth=azimuth_in_position,
    elevation=elevation_in_position,
    ... # do NOT end with force_output=True
)
```

Reply:
Added some more subtests for testing if moving to the same position. Updated the CSC to handle this case. Also changed moving to in_position as that more accurately reflects what the information received.